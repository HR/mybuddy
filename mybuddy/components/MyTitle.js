import React, { Component } from 'react'
import { Title } from 'native-base'
import PropTypes from 'prop-types'

export default class MyTitle extends Component {
  render () {
    const { style, ...props } = this.props
    return (
      <Title
        {...props}
        style={[
          {
            fontSize: 25,
            marginTop: 10,
            marginBottom: 5,
            color: 'black'
          },
          style
        ]}
      >
        {this.props.children}
      </Title>
    )
  }
}

MyTitle.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.string, PropTypes.node)
  ]).isRequired,
  style: PropTypes.object
}
