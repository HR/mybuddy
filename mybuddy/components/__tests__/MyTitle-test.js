import 'react-native'
import React from 'react'
import MyTitle from '../MyTitle'
import renderer from 'react-test-renderer'
import { Text } from 'native-base'

describe('My title rendering', () => {
  it('My title bar renders correctly', () => {
    expect(
      renderer
        .create(
          <MyTitle>
            <Text>test</Text>
          </MyTitle>
        )
        .toJSON()
    ).toMatchSnapshot()
  })
})
