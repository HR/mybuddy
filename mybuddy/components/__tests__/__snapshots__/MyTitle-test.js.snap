// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My title rendering My title bar renders correctly 1`] = `
<Text
  numberOfLines={1}
  style={
    Array [
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 17,
        "fontWeight": "700",
        "marginLeft": undefined,
        "paddingLeft": 4,
        "paddingTop": 1,
        "textAlign": "center",
      },
      Object {
        "color": "black",
        "fontSize": 25,
        "marginBottom": 5,
        "marginTop": 10,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
      }
    }
    uppercase={false}
  >
    test
  </Text>
</Text>
`;
